rust/Cargo.toml

```diff
@@ -5,8 +5,45 @@ edition = "2021"
 
 [dependencies]
 bnum = "0.11.0"
-evmc-vm = { git = "https://github.com/LorenzSchueler/evmc", branch = "tosca-extensions" } # TODO use Fantom-foundation/evmc once merged
+evmc-vm = { git = "https://github.com/Fantom-foundation/evmc", branch = "tosca-extensions" } # TODO use Fantom-foundation/evmc once merged
 sha3 = "0.10.8"
 
 [lib]
 crate-type = ["cdylib"]
+
+[lints.rust]
+macro_use_extern_crate = "warn"
+unused_crate_dependencies = "warn"
+unused_unsafe = "warn"
+
+[lints.clippy]
```

Comment:
Can you explain to us why we need to list the lints ?

Are there reduced in criticality? error to warning. or they are they turned on for the first time? 

I think this list is quite difficult to maintain, and I would rather prefer to have curated groups or defaults.
For example, can we add full grops from these categories?  https://doc.rust-lang.org/clippy/lints.html

The maintenance of this list requires an of investment, and it may not be so easy to find time for this in the future. For this reason this list may end up unmaintained, out of date, and inducing confusion.

Reply:
Those are all enabled for the first time. Most of them are in the `pendantic` lints group.

docs:
The clippy::pedantic group makes Clippy even more pedantic. You can enable the whole group with #![warn(clippy::pedantic)] in the lib.rs/main.rs of your crate. This lint group is for Clippy power users that want an in depth check of their code.

Note: Instead of enabling the whole group (like Clippy itself does), you may want to cherry-pick lints out of the pedantic group.

If you enable this group, expect to also use #[allow] attributes generously throughout your code. Lints in this group are designed to be pedantic and false positives sometimes are intentional in order to prevent false negatives.

If at any point we no longer want to use them we can just remove them and everthing keeps working. What should we do?

Reply:
ok, let's keep it then